### PR TITLE
BAN-1749 Change application header to X-Application-Key

### PR DESCRIPTION
**Pull request by matthieu526**
I jumped on the occasion to do a bit of code cleaning while I was working on this.

### CODE REVIEW

File: developer.go [hunk 1]
```diff
@@ -41,11 +42,12 @@ type DevClient struct {
 }
 
 // NewDevClient creates a new developer client, ready to use.
-func NewDevClient(client *http.Client, addr string, token string) *DevClient {
+func NewDevClient(client *http.Client, addr string, token string, applicationKey string) *DevClient {
```
Comment (denwwer) [inline developer.go]: "dev" requests is not required to have app key, only for user related data

File: service.go [hunk 2]
```diff
@@ -42,21 +42,23 @@ const (
...
+	environment    string
+	retryPolicy    RetryPolicy
 }
 
 type ClientOption func(*Client)
 
 // New creates a new client that will use the supplied HTTP client and connect
 // via the specified API host address.
-func New(client *http.Client, addr string, opts ...ClientOption) *Client {
+func New(client *http.Client, addr string, applicationKey string, opts ...ClientOption) *Client {
```
Comment (denwwer) [inline service.go]: is not required to have app key at start point

File: developer.go (hunk 1 above)
Comment (matthieu526) [reply developer.go]: I was wondering. So I should have let it in function parameters ?

File: service.go (hunk 2 above)
Comment (matthieu526) [reply service.go]: It is if we want to pass it along and have it.

File: app.go [hunk 3]
```diff
@@ -61,8 +59,8 @@ func (a *AppClient) newReq(path string) req {
 		addr: a.addr,
 		path: path,
 		headers: headers{
-			"User-Agent":       a.userAgent(),
-			"x-application-id": a.applicationID,
+			"User-Agent":        a.userAgent(),
+			"x-application-key": a.applicationKey,
```
Comment (gmonnier) [inline app.go]: Is this a breaking change? By this I mean will the old client still work with the applictaion ID header for at least a short while?

File: developer.go (hunk 1 above)
Comment (denwwer) [reply developer.go]: I guess yes

File: service.go (hunk 2 above)
Comment (denwwer) [reply service.go]: you can add some method to `Client ` for assign key

File: developer.go (hunk 1 above)
Comment (denwwer) [reply developer.go]: or have method to add it

File: service.go (hunk 2 above)
Comment (matthieu526) [reply service.go]: Not sure why we should do this in this case 🤔

File: developer.go (hunk 1 above)
Comment (matthieu526) [reply developer.go]: I think I'll let it like it was before

File: app.go (hunk 3 above)
Comment (gmonnier) [reply app.go]: ok got a talk with Alexis actually. LGTM

Comment (matthieu526) [reply app.go]: Not a breaking change. Except if we deploy our version before I updated the lib

File: service.go (hunk 2 above)
Comment (denwwer) [reply service.go]: yeah it only pass it to dev, so probably keep old version

**Pull request merged**